src/valkeymodule.h

```diff
@@ -0,0 +1,1711 @@
+    VALKEYMODULE_GET_API(UnregisterCommandFilter);
+    VALKEYMODULE_GET_API(CommandFilterArgsCount);
+    VALKEYMODULE_GET_API(CommandFilterArgGet);
+    VALKEYMODULE_GET_API(CommandFilterArgInsert);
+    VALKEYMODULE_GET_API(CommandFilterArgReplace);
+    VALKEYMODULE_GET_API(CommandFilterArgDelete);
+    VALKEYMODULE_GET_API(CommandFilterGetClientId);
+    VALKEYMODULE_GET_API(Fork);
+    VALKEYMODULE_GET_API(SendChildHeartbeat);
+    VALKEYMODULE_GET_API(ExitFromChild);
+    VALKEYMODULE_GET_API(KillForkChild);
+    VALKEYMODULE_GET_API(GetUsedMemoryRatio);
+    VALKEYMODULE_GET_API(MallocSize);
+    VALKEYMODULE_GET_API(MallocUsableSize);
+    VALKEYMODULE_GET_API(MallocSizeString);
+    VALKEYMODULE_GET_API(MallocSizeDict);
+    VALKEYMODULE_GET_API(CreateModuleUser);
+    VALKEYMODULE_GET_API(FreeModuleUser);
+    VALKEYMODULE_GET_API(SetContextUser);
+    VALKEYMODULE_GET_API(SetModuleUserACL);
+    VALKEYMODULE_GET_API(SetModuleUserACLString);
+    VALKEYMODULE_GET_API(GetModuleUserACLString);
+    VALKEYMODULE_GET_API(GetCurrentUserName);
+    VALKEYMODULE_GET_API(GetModuleUserFromUserName);
+    VALKEYMODULE_GET_API(ACLCheckCommandPermissions);
+    VALKEYMODULE_GET_API(ACLCheckKeyPermissions);
+    VALKEYMODULE_GET_API(ACLCheckChannelPermissions);
+    VALKEYMODULE_GET_API(ACLAddLogEntry);
+    VALKEYMODULE_GET_API(ACLAddLogEntryByUserName);
+    VALKEYMODULE_GET_API(DeauthenticateAndCloseClient);
+    VALKEYMODULE_GET_API(AuthenticateClientWithACLUser);
+    VALKEYMODULE_GET_API(AuthenticateClientWithUser);
+    VALKEYMODULE_GET_API(RedactClientCommandArgument);
+    VALKEYMODULE_GET_API(GetClientCertificate);
+    VALKEYMODULE_GET_API(GetCommandKeys);
+    VALKEYMODULE_GET_API(GetCommandKeysWithFlags);
+    VALKEYMODULE_GET_API(GetCurrentCommandName);
+    VALKEYMODULE_GET_API(RegisterDefragFunc);
+    VALKEYMODULE_GET_API(DefragAlloc);
+    VALKEYMODULE_GET_API(DefragRedisModuleString);
+    VALKEYMODULE_GET_API(DefragShouldStop);
+    VALKEYMODULE_GET_API(DefragCursorSet);
+    VALKEYMODULE_GET_API(DefragCursorGet);
+    VALKEYMODULE_GET_API(GetKeyNameFromDefragCtx);
+    VALKEYMODULE_GET_API(GetDbIdFromDefragCtx);
+    VALKEYMODULE_GET_API(EventLoopAdd);
+    VALKEYMODULE_GET_API(EventLoopDel);
+    VALKEYMODULE_GET_API(EventLoopAddOneShot);
+    VALKEYMODULE_GET_API(RegisterBoolConfig);
+    VALKEYMODULE_GET_API(RegisterNumericConfig);
+    VALKEYMODULE_GET_API(RegisterStringConfig);
+    VALKEYMODULE_GET_API(RegisterEnumConfig);
+    VALKEYMODULE_GET_API(LoadConfigs);
+    VALKEYMODULE_GET_API(RdbStreamCreateFromFile);
+    VALKEYMODULE_GET_API(RdbStreamFree);
+    VALKEYMODULE_GET_API(RdbLoad);
+    VALKEYMODULE_GET_API(RdbSave);
+
+    if (RedisModule_IsModuleNameBusy && RedisModule_IsModuleNameBusy(name)) return VALKEYMODULE_ERR;
+    RedisModule_SetModuleAttribs(ctx,name,ver,apiver);
+    return VALKEYMODULE_OK;
+}
+
+#define RedisModule_Assert(_e) ((_e)?(void)0 : (RedisModule__Assert(#_e,__FILE__,__LINE__),exit(1)))
```

Comment:
```suggestion
#define ValkeyModule_Assert(_e) ((_e)?(void)0 : (RedisModule__Assert(#_e,__FILE__,__LINE__),exit(1)))
```

Not sure if you meant to cover this define or meant to hit it later.

Reply:
yeah - I was planning to do just one PR for everything but it became big and then I changed my mind and was planning to stage the changes in multiple PRs but I guess that also creates this confusion about the final goal. I guess I will keep patching up this PR then.